validator.rb

```diff
@@ -112,17 +114,26 @@ def check_warnings(json)
         cleaned = cleaned.gsub(/ in schema (.*)/, '.')
         @errors << FeedError.new("error", cleaned)
       end
-      
+
       warnings = check_warnings(response_json)
       for w in warnings
         @errors << FeedError.new("warning", w)
-      end      
+      end
     rescue JSON::ParserError => e
       @errors << FeedError.new("error", "JSON could not be parsed. #{e.message}.")
     rescue FeedError => e
       @errors << e
     end
   end
-  
-  erb :index
+
+  case format
+  when "json"
+    content_type :json
+    {
+      valid: (@url.length > 0 && @errors.size == 0),
+      errors: @errors.map { |error| { error.type.capitalize.to_s => e.message } }
+    }.to_json
+  else
+    erb :index
+  end
```

Comment:
This new bit here will make the server respond in JSON format, if one added `format=json` as a query parameter. I copied the conditions and error key-value pairs from `index.erb` to be consistent with the HTML response. This will set the `Content-Type` HTTP header to `application/json` and render something like this (in a "valid" case):

```json
{
  "valid": true,
  "errors": []
}
```

Reply:
I thought about adding an `"url"` key as well to explicitly list the validated feed's URL. But for now decided against this. But this is something that might be useful.